src/pages/portfolio/[slug].astro

```diff
@@ -0,0 +1,77 @@
+---
+// import Layout from "../../layouts/Layout.astro";
+import Header from "../../components/Header.astro";
+import { SITE_TITLE, SITE_DESCRIPTION } from "../../consts";
+import Footer from "../../components/Footer.astro";
+import { getCollection, type CollectionEntry } from "astro:content";
+import BaseHead from "../../components/BaseHead.astro";
+import CardWrapper from "../../components/CardWrapper.astro";
+import Prose from "../../components/Prose.astro";
+import CollaborateCard from "../../components/CollaborateCard.astro";
+import { ViewTransitions } from "astro:transitions";
+
+// 1. Generate a new path for every collection entry
+export async function getStaticPaths() {
+  const PortfolioEntries = await getCollection("portfolio");
+  return PortfolioEntries.map((entry) => ({
+    params: { slug: entry.slug },
+    props: { entry },
+  }));
+}
+
+type Props = {
+  entry: CollectionEntry<"portfolio">;
+};
+
+// 2. For your template, you can get the entry directly from the prop
+const { entry } = Astro.props;
+const { Content } = await entry.render();
+---
+
+<html lang="en">
+  <head>
+    <BaseHead title={SITE_TITLE} description={SITE_DESCRIPTION} />
+    <ViewTransitions />
```

Comment:
Since we are going to want to use transitions everywhere on the site, lets import this in the BaseHead component